pkg/kademlia/routing_helpers.go

```diff
@@ -182,17 +143,18 @@ func (rt RoutingTable) putNode(nodeKey storage.Key, nodeValue storage.Value) err
 }
 
 // createOrUpdateKBucket: helper, adds or updates given kbucket
-func (rt RoutingTable) createOrUpdateKBucket(bucketID storage.Key, now time.Time) error {
-	dateTime := string(now.UTC().UnixNano())
+func (rt *RoutingTable) createOrUpdateKBucket(bucketID storage.Key, now time.Time) error {
+	dateTime := strconv.FormatInt(now.UTC().UnixNano(), 10)
```

Comment:
an alternate way is 

```
var dateType [8]byte
binary.BigEndian(dateType[:], now.UnixNano())
err := rt.kadBucketDB.Put(bucketID, dateType[:])
if err != nil {
   ...
```

encoding/binary is much more efficient in speed and space (and stores the number in binary form instead of base 10). you also don't need to convert a time to UTC before getting unix time. unix time is defined as in utc already. https://play.golang.org/p/rEYcjromLjk

Reply:
oh cool, good to know, thanks!